main.cpp

```diff
@@ -81,20 +95,37 @@ int main()
         window.draw(sprites["background"]);
         if (game == 1)
         {
+            if (!snake.isPosValid())
+            {
+                looseGame.stop();
+                looseGame.play();
+                game = 0;
+                continue;
+            }
+
             if (clock.getElapsedTime().asMilliseconds() >= SPEED)
             {
                 clock.restart();
                 moveSide = preMoveSide;
                 snake.move(moveSide);
-                if (!snake.inField())
-                {
-                    game = 0;
-                }
             }
         }
+        if (sprites["snakeHead"].getGlobalBounds() == sprites["apple"].getGlobalBounds())
+        {
+            lootApple.stop();
+            lootApple.play();
+            apple.create();
+            snake.increase();
+            if (snake.inApple())
```

Comment:
I propose to generate apple till it is in snake 
```suggestion
            while !(snake.inApple())
```

Reply:
i can't uderstand fully how its been working, give one more comment pls

Reply:
I forget operator not `!`
That gives possibility to convert true to false and false to true

I need to ensure apple is not in snake. So you do:

`apple.isInSnake(snake)` 
It returns true if apple in case 

You need to generate again apple coords, they need to be out of snake. So you check it again and again while it not outside 
`
apple.create()
if (apple.isInSnake(snake)) {
apple.create(); 
if (apple.isInSnake(snake)) {
apple.create(); 
if (apple.isInSnake(snake)) {
apple.create(); 
if (apple.isInSnake(snake)) {
apple.create(); 
if (apple.isInSnake(snake)) {
apple.create(); 
if (apple.isInSnake(snake)) {
apple.create(); 

} 
} 
} 
} 
} 
} 
`
Or easier with while or do-while operator
`
do {
apple.create();
} while !(apple.isInSnake(snake));
`

Reply:
super, thanks